### ISSUE TITLE
Reading metadata from xpt file in flask server

### ISSUE THREAD

**Issue by jvssago** [Roche/pyreadstat#108]
Hi, 

I am trying to read metadata from the xpt file using pyreadstat. while I am trying in my local system it is working fine. but when I am trying in Python Flask Server it is throwing exception like 'Unable to open file'.

Note: Even in the Flask server if I used a static absolute path it is working fine, but the same path if I send dynamically it is throwing an error.

Please find below the metadata reading sample.
    def prepare_metadata(self, path):
        try:
            m = {}
            df, meta = pyreadstat.read_xport(filename_path=path, metadataonly=True)
            print(meta.column_names_to_labels)
            for k, v in meta.column_names_to_labels.items():
                if not m.get(k) is None:
                    m[k].update({"label": v})
                else:
                    m.update({k: {"label": v}})
            for k, v in meta.variable_storage_width.items():
                if not m.get(k) is None:
                    m[k].update({"size": v})
                else:
                    m.update({k: {"size": v}})
            print("m:", m)
            return json.dumps(m)
        except Exception as e:
            print("prepare_metadata_err:", str(e))

**Comment by ofajardo** [comment]
hi

Since it is working fine in local, I am pretty sure this is a Flask question and not a pyreadstat issue; therefore I am going to close it.

I am no flask expert. I recently did a flask app where the user uploaded a csv, I then process it and spit out an excel. What I found is that flask handled all paths relative to the "app" folder. so I had a subfolder uploads (app/uploads) where I was saving the csvs, the path I had to use is relative to app, meaning uploads/file1.csv. In my hands giving any other kind of paths failed.

Now, pyreadstat is different, it uses in its backend a C library that knows nothing about the "app" folder. It will understand only two things:
* Either an absolute path and as you said that is working, so that is one possible solution, Instead of hard-coding you could do something like [finding out what is the path](https://stackoverflow.com/questions/595305/how-do-i-get-the-path-of-the-python-script-i-am-running-in) in one of your scripts (config.py for example). That's what I would do although I don't know what kind of security risks that brings.
* A relative path, but not relative to "app" but relative to where the process is being run. Where is that? depends on how you are running your server. If you would for example be running a dev server one level above app and your file would be in the upload subfolder of app then it would be "app/upload/file1.csv". But if you are running it in a different way, let's say another webserver like ngix, apache, or whatever, then I don't know, that a question where the answer depends on how you have configured it, and I cannot provide help there. Maybe print it somewhere to find out where you are, but probably it's better that you just use the absolute path.

Bottom line, you need to pass an absolute path to pyreadstat or to find out where your process is running in order to pass an appropiate relative path and that behavior cannot be changed as it comes from the C library. How to better handle this path thing in flask is a flask/webserver question and not a pyreadstat issue.

Hope that helps.

**Comment by ofajardo** [comment]
or maybe it is the other way around regarding relative paths and indeed all relative paths are given relative to "app" folder and you have to calculate where your xpt file is respective to that. Not sure.

**Comment by jvssago** [comment]
Thanks for quick response.

I will check the path issue and let you know
